include/mrdox/Support/Handlebars.hpp

```diff
@@ -0,0 +1,1037 @@
+    }
+
+    // stream to output
+    static
+    void
+    noop_output(void *, std::string_view) {}
+
+    // Noop constructor
+    // Used as implementation detail by Handlebars engine
+    OutputRef()
+        : out_( nullptr )
+        , fptr_( &noop_output )
+    {}
+
+public:
+    /** Constructor for std::string output
+
+        @param st The string to append to
+     */
+    template<SVAppendable St>
+    requires std::same_as<typename St::value_type, char>
+    OutputRef( St& st )
+      : out_( &st )
+      , fptr_( &append_to_output<St> )
+    {
+    }
+
+    /** Constructor for llvm::raw_string_ostream output
+
+        @param os The output stream to write to
+     */
+    template <StdLHROStreamable Os>
+    requires std::is_convertible_v<Os*, std::ostream*>
+    OutputRef( Os& os )
+        : out_( &os )
+        , fptr_( &write_to_output<Os> )
+    {
+    }
+
+    /** Constructor for std::ostream& output
+
+        @param os The output stream to write to
+     */
+    template <LHROStreamable Os>
+    requires std::is_convertible_v<Os*, std::ostream*>
+    OutputRef( Os& os )
+        : out_( &os )
+        , fptr_( &write_to_output<Os> )
+    {
+    }
+
+    /** Write to output
+
+        @param sv The string to write
+        @return A reference to this object
+     */
+    OutputRef&
+    operator<<( std::string_view sv )
+    {
+        fptr_( out_, sv );
+        return *this;
+    }
+
+    /** Write to output
+
+        @param c The character to write
+        @return A reference to this object
+     */
+    OutputRef&
+    operator<<( char c )
+    {
+        fptr_( out_, std::string_view( &c, 1 ) );
+        return *this;
+    }
+
+    /** Write to output
+
+        @param c The character to write
+        @return A reference to this object
+     */
+    template <class T>
+    requires fmt::has_formatter<T, fmt::format_context>::value
+    OutputRef&
+    operator<<( T v )
+    {
+        std::string s = fmt::format( "{}", v );
+        fptr_( out_, s );
+        return *this;
+    }
+};
+
+/** Callback information for handlebars helpers
+
+    This class is used to pass information about the current
+    context to handlebars helpers.
+
+    It allows the helpers to access the current context,
+    the current output stream, and render the current block.
+
+ */
+class MRDOX_DECL HandlebarsCallback
+{
+private:
+    std::function<void(OutputRef, dom::Value const&)> fn_;
+    std::function<void(OutputRef, dom::Value const&)> inverse_;
+    dom::Value const* context_{ nullptr };
+    OutputRef* output_{ nullptr };
+    dom::Object data_;
+    dom::Object hashes_;
+    dom::Array blockParams_;
+    friend class Handlebars;
+    HandlebarsCallback();
+
+public:
+    /** Render the block content with the specified context
+
+        This function renders the block content with the specified context.
+
+        For example, the following template calls the helper "list" with a block section:
+
+        @code{.handlebars}
```

Comment:
Does this work? .handlebars ?

Reply:
In doxygen HTML `{.handlebars}` is basically just indicating this is not `cpp` because it can't highlight handlebars.

> If the contents of the code block are in a language that doxygen cannot parse, doxygen will just show the output as-is. You can make this explicit using .unparsed, or by giving some other extension that doxygen doesn't support, e.g.

Outside the context of doxygen or when using doxygen XML to generate something else, then it can be useful to render it as a handlebars source block instead of C++. The logic needs to be implemented but the information is there.